src/BoundaryConditions.h

```diff
@@ -177,7 +177,7 @@ inline NO_INLINE Func repeat_image(const T &func_like) {
 
 template <typename T, typename ...Bounds,
           typename std::enable_if<Halide::Internal::all_are_convertible<Expr, Bounds...>::value>::type* = nullptr>
-inline NO_INLINE Func repeat_image(const T &func_like, Bounds&&... bounds) {
+inline NO_INLINE Func repeat_image(T &func_like, Bounds&&... bounds) {
```

Comment:
In case you're wondering, these things all became non-const because if a Pipeline captures an Buffer by reference, it might actually mutate the Buffer via that reference (e.g. to give it a device allocation).